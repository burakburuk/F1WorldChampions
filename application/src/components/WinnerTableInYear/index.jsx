import React from 'react';
import PropTypes from 'prop-types';

import {
    Table,
    TableHead,
    TableRow,
    TableBody,
    TableCell,
} from '../Table';
import StyledWinnerTableInYear from './StyledWinnerTableInYear';

const WinnerTableInYear = props => (
    <StyledWinnerTableInYear>
        <Table>
            <TableHead>
                <TableRow>
                    <TableCell>Race</TableCell>
                    <TableCell>Driver</TableCell>
                    <TableCell>Date</TableCell>
                    <TableCell>Team</TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {props.winners}
            </TableBody>
        </Table>
    </StyledWinnerTableInYear>
);

WinnerTableInYear.propTypes = {
    winners: PropTypes.array,
};

export default WinnerTableInYear;
