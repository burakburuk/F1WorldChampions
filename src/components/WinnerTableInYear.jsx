import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
    Table,
    TableHead,
    TableRow,
    TableBody,
    TableCell,
} from './Table';

const StyledWinnerTableInYear = styled.div`
    width: 100%;
    padding-top: 10px;
    padding-left: 30px;
    padding-right: 30px;

    td {
        border-bottom: 1px solid #9e9e9e;
        height: 50px;
    }
`;

const WinnerTableInYear = props => (
    <StyledWinnerTableInYear>
        <Table>
            <TableHead>
                <TableRow>
                    <TableCell>Race</TableCell>
                    <TableCell>Driver</TableCell>
                    <TableCell>Date</TableCell>
                    <TableCell>Team</TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {props.winners}
            </TableBody>
        </Table>
    </StyledWinnerTableInYear>
);

WinnerTableInYear.propTypes = {
    winners: PropTypes.array,
};

export default WinnerTableInYear;
