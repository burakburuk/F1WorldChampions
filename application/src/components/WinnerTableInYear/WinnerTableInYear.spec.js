import React from 'react';
import { shallow } from 'enzyme';
import WinnerTableInYear from '.';

it('should match with snapshot', () => {
    const wrapper = shallow(<WinnerTableInYear />);
    expect(wrapper).toMatchSnapshot();
});
