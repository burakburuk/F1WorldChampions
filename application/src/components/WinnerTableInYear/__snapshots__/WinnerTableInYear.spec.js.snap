// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<styled.div>
  <Table>
    <TableHead>
      <TableRow>
        <TableCell>
          Race
        </TableCell>
        <TableCell>
          Driver
        </TableCell>
        <TableCell>
          Date
        </TableCell>
        <TableCell>
          Team
        </TableCell>
      </TableRow>
    </TableHead>
    <TableBody />
  </Table>
</styled.div>
`;
